### PR TITLE
Add container-allowed-capabilities PSP

### PR DESCRIPTION
**Pull request by EmandM**

### CODE REVIEW

File: Policies/Kubernetes/container-allowed-capabilities/azurepolicy.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        "in": [
+          "AKS Engine",
+          "Microsoft.Kubernetes/connectedClusters",
+          "Microsoft.ContainerService/managedClusters"
+        ]
+      },
+      "then": {
+        "effect": "[parameters('effect')]",
+        "details": {
+          "constraintTemplate": "https://raw.githubusercontent.com/EmandM/Community-Policy/psp/capabilities/Policies/Kubernetes/container-allowed-capabilities/template.yaml",
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-allowed-capabilities/azurepolicy.json:62]: Update template and constraint URLs before you check-in.

File: Policies/Kubernetes/container-allowed-capabilities/azurepolicy.rules.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "in": [
+      "AKS Engine",
+      "Microsoft.Kubernetes/connectedClusters",
+      "Microsoft.ContainerService/managedClusters"
+    ]
+  },
+  "then": {
+    "effect": "[parameters('effect')]",
+    "details": {
+      "constraintTemplate": "https://raw.githubusercontent.com/EmandM/Community-Policy/psp/capabilities/Policies/Kubernetes/container-allowed-capabilities/template.yaml",
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-allowed-capabilities/azurepolicy.rules.json:13]: Update links here too.

File: Policies/Kubernetes/container-allowed-capabilities/rego/src_test.rego [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+   input := { "review": input_review([cdrop(["one"]), cdrop(["one", "two"])]), "parameters": {"requiredDropCapabilities": ["one", "two"]}}
+   results := violation with input as input
+   count(results) == 1
+}
+test_input_missing_drop_x2 {
+   input := { "review": input_review([cdrop(["one"]), cdrop(["two"])]), "parameters": {"requiredDropCapabilities": ["one", "two"]}}
+   results := violation with input as input
+   count(results) == 2
+}
+test_input_drop_undefined_x2 {
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-allowed-capabilities/rego/src_test.rego:101]: Can you please add a test for input drop undefined and requiredDropCapabilities is empty array?

File: Policies/Kubernetes/container-allowed-capabilities/rego/src_test.rego [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    input := { "review": input_review([cadd(["one"]), cadd(["two"])]), "parameters": {"allowedCapabilities": ["one", "two"]}}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_two_allowed_two_used_container_x2 {
+    input := { "review": input_review([cadd(["one", "two"]), cadd(["one", "two"])]), "parameters": {"allowedCapabilities": ["one", "two"]}}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_none_allowed {
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-allowed-capabilities/rego/src_test.rego:33]: Can you please add below two tests?
1. add capabilities not defined in input object and allowedCapabilities array is also empty
2. add capabilities not defined in input object and allowedCapabilities array is not empty

File: Policies/Kubernetes/container-allowed-capabilities/azurepolicy.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
+{
+  "properties": {
+    "displayName": "Ensure only allowed capabilities are used in Kubernetes Cluster",
+    "policyType": "Custom",
+    "mode": "Microsoft.Kubernetes.Data",
+    "description": "This policy ensures only given capabilities are used inside a Kubernetes cluster. For instructions on using this policy, please visit https://aka.ms/kubepolicydoc.",
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-allowed-capabilities/azurepolicy.json:6]: This policy ensures only specified capabilities are defined on containers in a Kubernetes cluster.

Comment (EmandM) [reply Policies/Kubernetes/container-allowed-capabilities/azurepolicy.json:6]: I've changed to `This policy ensures specified security capabilities are defined and removed inside a Kubernetes cluster. For instructions on using this policy, please visit https://aka.ms/kubepolicydoc.`, Happy to change it to this here, however I feel this description misses the functionality of the required drop capabilities

Comment (RamyasreeChakka) [reply Policies/Kubernetes/container-allowed-capabilities/azurepolicy.json]: I would mention containers explicitly somewhere in description as capabilities are defined on containers.
